packages/react-interactions/events/src/dom/create-event-handle/__tests__/useFocus-test.internal.js

```diff
@@ -91,7 +91,7 @@ describe.each(table)(`useFocus`, hasPointerEvents => {
         useFocus(ref, {
           onBlur,
         });
-        return <div ref={ref} />;
+        return <div ref={ref} onBlur={() => {}} onFocus={() => {}} />;
```

Comment:
I don't understand why we need this. The same fix that worked for `useFocusWithin` did not work for `useFocus`. It only registers these events if the target also has corresponding event handlers attached.

It might be related to the focus target which is an <input /> in `useFocusWithin` tests but a generic <div> (which is technically not focusable in a browser) in `useFocus` tests

Reply:
Looks like this is indeed an issue with the target:
```jsx
// @gate experimental
it('works on an <input /> that does not have onFocus listener', () => {
  const onFocus = jest.fn();
  const ref = React.createRef();
  const Component = () => {
    useFocus(ref, {
      onFocus,
    });
    return <input ref={ref} />;
  };
  ReactDOM.render(<Component />, container);
  Scheduler.unstable_flushAll();

  const target = createEventTarget(ref.current);
  target.focus();

  expect(onFocus).toHaveBeenCalledTimes(1);
});
```
works without having to attach an empty `onFocus` listener.

Reply:
From what I can gather having an `<input />` means that you have an implicit `onFocus` handler due to the `ChangeEvent` or `BeforeInput` plugins. On a `button` React will not attach additional focus handlers which, for some reason or another, leads to `useFocus` not working.

Reply:
So I don't quite understand why `<input />` works but `<button />` doesn't but there is a "defect" that causes `<button />` to not work: In https://github.com/facebook/react/blob/b61174fb7b09580c1ec2a8f55e73204b706d2935/packages/react-dom/src/events/DOMPluginEventSystem.js#L733 we check if a given React event.type is equal to a `DOMEventName` from `ReactDOMEventHandleListener`. It's basically assuming that the Synthetic event.type is equal to the native event type.

If we go from change 
```diff
-const targetType = event.type; 
+const targetType = event.nativeEvent.type; 
```
in https://github.com/facebook/react/blob/b61174fb7b09580c1ec2a8f55e73204b706d2935/packages/react-dom/src/events/DOMPluginEventSystem.js#L712
then the `useFocus` tests are passing: [`cc2ae0c` (#19561)](https://github.com/facebook/react/pull/19561/commits/cc2ae0ca2890de9b93674f5671300e0f2c80567f)